cpp/src_prims/score/scores.h

```diff
@@ -21,14 +21,176 @@
 #include "linalg/subtract.h"
 #include "linalg/power.h"
 
+#include <memory>
+
 #include "common/cuml_allocator.hpp"
 
+#include "selection/knn.h"
+#include "distance/distance.h"
+#include <selection/columnWiseSort.h>
+
+
 #include <thrust/reduce.h>
 #include <thrust/device_ptr.h>
 
+#define MAX_BATCH_SIZE 512
+#define N_THREADS 512
+
+
 namespace MLCommon {
     namespace Score {
-        template<typename math_t>
+
+      /**
+      * @brief Compute a the rank of trustworthiness score
+      * @input param ind_X: indexes given by pairwise distance and sorting
+      * @input param ind_X_embedded: indexes given by KNN
+      * @input param n: Number of samples
+      * @input param n_neighbors: Number of neighbors considered by trustworthiness score
+      * @input param work: Batch to consider (to do it at once use n * n_neighbors)
+      * @output param rank: Resulting rank
+      */
+      template<typename math_t>
+      __global__ void compute_rank(math_t *ind_X, long *ind_X_embedded,
```

Comment:
question: why is `ind_X` `math_t` while `ind_X_embedded` a `long`?

Reply:
Unfortunately, FAISS hardcodes the resulting index matrix to a `long` so this is out of our control. That said, I'm totally okay introducing a separate parameter to capture possible differences in implementations between the pairwise distance and knn.